includes/rest-api/controllers/class-feedback-controller.php

```diff
@@ -0,0 +1,189 @@
+	 * @since [next-version-number]
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => \WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'upsert_feedback' ),
+					'permission_callback' => array( $this, 'create_or_update_feedback_permissions_check' ),
+				),
+			)
+		);
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<survey_id>\d+)',
+			array(
+				array(
+					'methods'             => \WP_REST_Server::EDITABLE,
+					'callback'            => array( $this, 'upsert_feedback' ),
+					'permission_callback' => array( $this, 'create_or_update_feedback_permissions_check' ),
+					'args'                => $this->get_feedback_fetch_params(),
+				),
+			)
+		);
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<survey_id>\d+)/response',
+			array(
+				array(
+					'methods'             => \WP_REST_Server::EDITABLE,
+					'callback'            => array( $this, 'upsert_feedback_response' ),
+					'permission_callback' => array( $this, 'create_or_update_feedback_response_permissions_check' ),
+					'args'                => $this->get_feedback_fetch_params(),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Updates an Feedback Survey. Creates one if no ID is given.
+	 *
+	 * @since [next-version-number]
+	 *
+	 * @param  \WP_REST_Request $request The API Request.
+	 * @return \WP_REST_Response|WP_Error
+	 */
+	public function upsert_feedback( \WP_REST_Request $request ) {
+		$data   = $request->get_json_params();
+		$survey = Feedback_Survey::from_block_attributes( $data );
+		$result = Crowdsignal_Forms::instance()->get_api_gateway()->update_feedback( $survey );
+
+		if ( is_wp_error( $result ) ) {
+			return $result;
+		}
+
+		return rest_ensure_response( $result->to_block_attributes() );
+	}
+
+	/**
+	 * This route acts as a proxy for Crowdsignal's Feedback response endpoint,
+	 * which allows recording and updating responses.
+	 *
+	 * @since [next-version-number]
+	 *
+	 * @todo The nonce helps but it's still possible for someone to generate their own nonce and
+	 *       submit someone else's response id.
+	 *       The nonce needs to be tied to the response ID.
+	 *
+	 * @param  \WP_REST_Request $request The API Request.
+	 * @return \WP_REST_Response|WP_ERROR
+	 */
+	public function upsert_feedback_response( \WP_REST_Request $request ) {
+		$data      = $request->get_json_params();
+		$survey_id = $request->get_param( 'survey_id' );
+
+		$verifies = Crowdsignal_Forms_Feedback_Block::verify_nonce( $data['nonce'] );
+
+		if ( ! $verifies ) {
+			return new \WP_Error( 'Forbidden' );
+		}
+
+		$result = Crowdsignal_Forms::instance()->get_api_gateway()->update_feedback_response(
+			$survey_id,
+			$data
+		);
+
+		if ( is_wp_error( $result ) ) {
+			return $result;
+		}
+
+		$result['checksum'] = $this->get_response_checksum( $result['r'], $data['nonce'] );
```

Comment:
I believe we submit feedback only once so we don't actually need the checksum to make sure both responses are coming from the same source.